lib/httpclient/http.rb

```diff
@@ -574,17 +574,19 @@ def reset_pos(io)
       end
 
       def dump_file(io, dev, sz)
+
         buf = ''
         rest = sz
         while rest > 0
           n = io.read([rest, @chunk_size].min, buf)
-          raise ArgumentError.new("Illegal size value: #size returns #{sz} but cannot read") if n.nil?
```

Comment:
This phenomenon occurred when I was testing with [webmock](https://github.com/bblimke/webmock).

I seem that the same proccessing is performed twice in a testing.

Maybe, around here file.
https://github.com/bblimke/webmock/blob/master/lib/webmock/http_lib_adapters/httpclient_adapter.rb